src/options.toml

```diff
@@ -34,6 +34,43 @@
 #All of these options are required
 #If they are not provided the program will exit early
 [BCs]
+    # Tells the program that we'll have changing BCs
+    # If left out of the file this defaults to a value of false
+    changing_ess_bcs = false
+    # The below gives an example of how to make use of changing essential BCs
+    # where we have the sample cyclically loaded.
+    # It would go in tandem with using the auto time stepper with a dt = 0.1
+    # and t_final = 7.0.
+    # The step number that a new BC is to be applied on.
+    # You must always have step 1 listed here, since that is the initial
+    # step that any BC is applied on.
+    # update_steps = [1, 11, 31, 51, 71]
+    # You'll want to have all the BC ids that you plan on using through out the
+    # simulation listed here. Also, when constraining the problem you'll want to
+    # take into consideration that currently we can't constrain a boundary to have
+    # a desired normal through out the deformation history.
+    #essential BC ids for the whole boundary
+    #essential_ids = [[1, 2, 3, 4],
+    #                 [1, 2, 3, 4],
+    #                 [1, 2, 3, 4],
+    #                 [1, 2, 3, 4],
+    #                 [1, 2, 3, 4]]
+    #component combo (x,y,z = -1, x = 1, y = 2, z = 3, xy = 4, yz = 5, xz = 6, free = 0)
+    #essential_comps = [[3, 1, 2, 3],
+    #                   [3, 1, 2, 3],
+    #                   [3, 1, 2, 3],
+    #                   [3, 1, 2, 3],
+    #                   [3, 1, 2, 3]]
+    #Vector of vals to be applied for each attribute
+    #The length of this should be #ids * dim of problem
+    #essential_vals = [[0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.001],
+    #                  [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, -0.001],
+    #                  [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.001],
+    #                  [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, -0.001],
+    #                  [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.001]]
```

Comment:
So if these are velocity values then you do not hold a truly steady velocity gradient. Would it make sense, as part of the changing-essential-BCs feature improvement to allow for a case in which you specify a velocity gradient and it uses the coordinates to compute the velocity for each time step? That might be a feature that folks would appreciate.

Reply:
So, I would say prescribing a constant velocity gradient through out the deformation of the simulation is not super simple outside of maybe uniaxial loading where it simplifies to something like having a constant strain rate. I might be able to support that uniaxial case, but I would probably want to do that after this PR. Since, it'll take me a little bit to make sure we're doing it right.

I imagine once we add support for periodic boundary conditions then we should also be able to add support for a constant velocity gradient to be applied.

Reply:
A simple implementation ought to be not much more than v_i = L_ij x_j with x being the current position of the given node. Or perhaps v=L.(x-x0) if you want to generalize so that the user need not have a corner of the domain at x=0. But it can certainly wait for a future PR.

Reply:
Yeah that might work assuming it's only done on the BCs and takes into account what dofs are constrained. I'll need to figure out a nice way to have it implemented, so we get out a v_i that corresponds to time: (t_i+1 + t_i) / 2.